pkg/utils/hash/hash.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2016 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hash
+
+import (
+	"crypto/sha512"
+	"fmt"
+)
+
+const maxLen = sha512.Size * 2
+
+func New(input string, length int) (string, error) {
+	if length <= 0 || length > maxLen {
+		return "", fmt.Errorf("only length between 1 and 128 are supported")
```

Comment:
Hmmm... IMO instead of 128 there should be used `maxLen` value, so if someday we would switch from sha512 to something other - this would be still correct.